AppController/djinn.rb

```diff
@@ -5669,159 +5758,153 @@ def try_to_scale_down(app_name, delta_appservers)
   # the old application code can be forced with a parameter.
   #
   # Args:
-  #   app       : the application name to setup
+  #   version_key: the version to setup
   #   remove_old: boolean to force a re-setup of the app from the tarball
-  def setup_app_dir(app, remove_old=false)
-    app_dir = "#{HelperFunctions.get_app_path(app)}/app"
-
+  def setup_app_dir(version_key, remove_old=false)
+    project_id, service_id, version_id = version_key.split('_')
     begin
       version_details = ZKInterface.get_version_details(
-        app, DEFAULT_SERVICE, DEFAULT_VERSION)
+        project_id, service_id, version_id)
     rescue VersionNotFound
       Djinn.log_debug(
-        "Skipping #{app} setup because version node does not exist")
+        "Skipping #{version_key} setup because version node does not exist")
       return
     end
 
-    app_path = version_details['deployment']['zip']['sourceUrl']
     error_msg = ""
 
-    if remove_old
-      Djinn.log_info("Removing old application version for app: #{app}.")
-      if my_node.is_shadow?
-        # Force the shadow node to refresh the application directory.
-        FileUtils.rm_rf(app_dir)
-      else
-        FileUtils.rm_rf(app_path)
-        revision_key = [app, DEFAULT_SERVICE, DEFAULT_VERSION,
-                        version_details['revision'].to_s].join('_')
-        stop_hosting_revision(revision_key, app_path, @@secret)
-      end
-    end
-
-    # Let's make sure we have a copy of the tarball of the application. If
-    # not, we will get the latest version from another node.
-    FileUtils.rm_rf(app_dir) unless File.exists?(app_path)
+    revision_key = [version_key, version_details['revision'].to_s].join('_')
+    if remove_old && my_node.is_load_balancer?
```

Comment:
It's not needed now because nginx is not being configured properly to serve static assets. I could remove it, but I think it's useful for associating that work with the load balancer role.